blaze-client/src/main/scala/org/http4s/client/blaze/Http1Connection.scala

```diff
@@ -380,7 +416,7 @@ private object Http1Connection {
 
   // ADT representing the state that the ClientStage can be in
   private sealed trait State
-  private case object Idle extends State
+  private final case class Idle(idleRead: Option[Future[ByteBuffer]]) extends State
```

Comment:
Is `Idle(None)` just an init state?  I might have made a separate `State` and eliminated the `Option`, but I guess we're treating both variants of `Idle` the same in lots of places.  Maybe my idea cleans it up, maybe it makes it worse.  I trust your judgment.

Reply:
Yes, it's just the state before running the first request. 

I just had a second look at that state and noticed I can remove two lines of code: ff0cc53 

I tried refactoring the code to have this separate state. This is what I came up with:
Option A: https://github.com/RafalSumislawski/http4s/commit/efb7afbb79eaeb252f0caabda0bb151921d6bc47
Option B: https://github.com/RafalSumislawski/http4s/commit/ef024654a38e0bb90cf998aab5a850897fbf129c
I like the fact that the additional state makes the model of the state machine more clear, but I don't like the additional LoC. Please let me know if I should bring one of these options into this PR. If I were to make the decision, I would choose the current shape of the PR.

Reply:
Thanks for trying.  I think you're right.  Let's keep it as is.